routing/pathfind.go

```diff
@@ -1007,18 +1007,22 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 				continue
 			}
 
-			firstHopTLVs := lnwire.CustomRecords(
-				r.FirstHopCustomRecords,
-			)
-			firstHopData, err := firstHopTLVs.Serialize()
-			if err != nil {
-				return nil, 0, err
+			var firstHopBlob fn.Option[tlv.Blob]
+			if r.FirstHopCustomRecords != nil {
```

Comment:
Perhaps should've just done `len(r.FirstHopCustomRecords) > 0` to be on the extra cautious side?